apsuite/orbcorr/orbit_correction.py

```diff
@@ -202,20 +213,34 @@ def _process_kicks(self, dkicks):
         que = [(-par.maxkickch - kickch) / dkickch, ]
         que.append((par.maxkickch - kickch) / dkickch)
         que = _np.max(que, axis=0)
-        dkickch *= min(_np.min(que), 1.0)
+        coef_ch = min(_np.min(que), 1.0)
 
         que = [(-par.maxkickcv - kickcv) / dkickcv, ]
         que.append((par.maxkickcv - kickcv) / dkickcv)
         que = _np.max(que, axis=0)
-        dkickcv *= min(_np.min(que), 1.0)
+        coef_cv = min(_np.min(que), 1.0)
 
+        coef_rf = 1.0
         if self.params.enblrf and dkickrf != 0:
             que = [(-par.maxkickrf - kickrf) / dkickrf, ]
             que.append((par.maxkickrf - kickrf) / dkickrf)
             que = _np.max(que, axis=0)
-            dkickrf *= min(_np.min(que), 1.0)
+            coef_rf = min(_np.min(que), 1.0)
+
+        min_coef = min(coef_ch, coef_cv, coef_rf)
+
+        if self.use_min_coef:
+            dkickch *= min_coef
+            dkickcv *= min_coef
+            dkickrf *= min_coef
+            saturation_flag = min_coef == 0
+        else:
+            dkickch *= coef_ch
+            dkickcv *= coef_cv
+            dkickrf *= coef_rf
+            saturation_flag = False
```

Comment:
I think in this case saturation_flag should be raised in case any of the coefficients are zero. You can accomplish this with:
```python3
if self.use_min_coef:
    dkickch *= min_coef
    dkickcv *= min_coef
    dkickrf *= min_coef
else:
    dkickch *= coef_ch
    dkickcv *= coef_cv
    dkickrf *= coef_rf
saturation_flag = _np.isclose(min_coef, 0)
```
I think it is more general to use `_np.isclose` than the equality in this case, since